src/TSMapEditor/UI/CursorActions/DrawCliffCursorAction.cs

```diff
@@ -0,0 +1,187 @@
+﻿using Microsoft.Xna.Framework;
+using Rampastring.XNAUI;
+using Rampastring.XNAUI.Input;
+using System;
+using System.Collections.Generic;
+using TSMapEditor.GameMath;
+using TSMapEditor.Models;
+using TSMapEditor.Models.Enums;
+using TSMapEditor.Mutations.Classes;
+using TSMapEditor.Rendering;
+
+namespace TSMapEditor.UI.CursorActions
+{
+    /// <summary>
+    /// Cursor action for placing bridges.
+    /// </summary>
+    public class DrawCliffCursorAction : CursorAction
+    {
+        public DrawCliffCursorAction(ICursorActionTarget cursorActionTarget, CliffType cliffType) : base(cursorActionTarget)
+        {
+            this.cliffType = cliffType;
+            ActionExited += UndoOnExit;
+        }
+
+        public override string GetName() => "Draw Cliff";
+
+        public override bool HandlesKeyboardInput => true;
+
+        public override bool DrawCellCursor => true;
+
+        private readonly CliffType cliffType;
+
+        private List<Point2D> cliffPath;
+        private CliffSide cliffSide = CliffSide.Front;
+        private DrawCliffMutation previewMutation;
+        private byte extraHeight = 0;
+
+        private readonly int randomSeed = Guid.NewGuid().GetHashCode();
```

Comment:
Not that it matters a lot, but this seems a bit overkill.

```suggestion
        private readonly int randomSeed = new Random().Next();
```